tests/integration/bond_test.py

```diff
@@ -553,3 +556,57 @@ def test_set_miimon_100_on_existing_bond(bond99_with_2_slaves):
     bond_config[Bond.OPTIONS_SUBTREE] = {"miimon": 100}
     libnmstate.apply(state)
     assertlib.assert_state_match(state)
+
+
+@pytest.fixture
+def eth1_eth2_with_no_profile():
+    iface_down(ETH1)
+    iface_down(ETH2)
+    yield
+    iface_down(ETH1)
+    iface_down(ETH2)
+
+
+def _nmcli_profile_reactivate(ifname):
+    """
+    Use nmcli to deactivate the profile and reactivate
+    """
+    assert cmdlib.exec_cmd(["nmcli", "connection", "down", ifname])[0] == 0
+    # Wait slave been deactivated
+    time.sleep(1)
+    assert cmdlib.exec_cmd(["nmcli", "connection", "up", ifname])[0] == 0
+    # Wait slave been activated
+    time.sleep(1)
+
+
+def test_bond_mac_be_consistent_with_diff_slave_order(
+    eth1_eth2_with_no_profile,
+):
+    """
+    When sever reboot, network manager will by default activate slaves in the
+    order of their name. Nmstate should provide the consistent MAC address for
+    bond regardless the order of slaves.
+    Use nmcli to reactivate the profile to simulate the server reboot.
+    """
+    eth1_mac = get_mac_address(ETH1)
+    with bond_interface(
+        name=BOND99,
+        slaves=[ETH2, ETH1],
+        extra_iface_state={
+            Bond.CONFIG_SUBTREE: {Bond.MODE: BondMode.ROUND_ROBIN}
+        },
+    ):
+        assert get_mac_address(BOND99) == eth1_mac
+        _nmcli_profile_reactivate(BOND99)
+        assert get_mac_address(BOND99) == eth1_mac
+
+    with bond_interface(
+        name=BOND99,
+        slaves=[ETH1, ETH2],
+        extra_iface_state={
+            Bond.CONFIG_SUBTREE: {Bond.MODE: BondMode.ROUND_ROBIN}
+        },
+    ):
+        assert get_mac_address(BOND99) == eth1_mac
+        _nmcli_profile_reactivate(BOND99)
```

Comment:
AFAIU, it might also happen during boot that a slave is activated before the bond which will then activate the bond + other slave. What happens then?

Reply:
The slave profile cannot be activated before master. And the master activation will bring up
all its slaves when `connection.autoconnect-slaves=1(yes)` which is what we set in nmstate.

Reply:
Do we have a test for this setting?

Reply:
No. It seems works well with no problems. I don't know how to test it.